File: 283-MoveZeroes/283-MoveZeroes.java
// Last updated: 7/28/2025, 12:30:41 PM
class Solution {
    public void moveZeroes(int[] nums) {
        int low = 0 , high = 1;
        int n = nums.length;
        while(high<n){
            if(nums[low]==0 && nums[high]!=0){
                int temp = nums[low];
                nums[low] = nums[high];
                nums[high] = temp; 
            }
            else if(nums[low] == 0){
                high++;
            }
            else {
                low++;
                high++;
            }
        }
    }
    
}

File: 283-MoveZeroes/283-MoveZeroes.cpp
// Last updated: 7/28/2025, 12:30:36 PM
class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        int i=0;
        int n = nums.size();
        for(int j=1;j<n;j++){
            
            if(nums[i] == 0 && nums[j]!=0){
                int temp = nums[i];
                nums[i] = nums[j];
                nums[j] = temp;
            
            }
            if(nums[i]!=0) i++;
        }
        
    }
};